src/components/playback/playqueuemanager.js

```diff
@@ -56,6 +58,36 @@ define([], function () {
         }
     };
 
+    PlayQueueManager.prototype.shufflePlaylist = function () {
+        this._sortedPlaylist = [];
+        let currentPlaylistItem = this._playlist[this.getCurrentPlaylistIndex()];
+        for (let item of this._playlist) {
+            this._sortedPlaylist.push(item);
+        }
+
+        for (let i = this._playlist.length - 1; i > 0; i--) {
+            if (this._playlist[i] === currentPlaylistItem) {
+                this._playlist.splice(i, 1);
+                continue;
+            }
+            const j = Math.floor(Math.random() * i);
+            const temp = this._playlist[i];
+            this._playlist[i] = this._playlist[j];
+            this._playlist[j] = temp;
+        }
+        this._playlist.unshift(currentPlaylistItem);
```

Comment:
@dmitrylyzo Yes, in the splice. Afterwards, with the continue, it goes to the following item.

After adding this I did a search multiple times and I couldn't find duplicate items. You're saying this because you actually had this happening?

Reply:
> You're saying this because you actually had this happening?

Yes.
I placed breakpoints in "remove current" block and before `unshift`.
It didn't stop in "remove current" block.
Before `unhisht`, I still have current item.

Fun fact, when current item is shuffled at the end of queue, the latter is virtually looped - player switches to "first" current item instead of shuffled last.

Reply:
@dmitrylyzo Check how I managed this in my next commits. I removed it before shuffling and then added it afterwards.

Reply:
It sounds promising.